Implement the NOTICE when REVOKE is a no-op differently.

hlinnaka opened the pull request:
See discussion at:

https://groups.google.com/a/greenplum.org/d/msg/gpdb-dev/Ttn_UJb4Otg/LS1cFrDiAwAJ

This gives a less verbose message that doesn't mention the object or role name. Also, if you list multiple roles or tables, you only get a NOTICE, if nothing was revoked from any of the objects. Currently you get a NOTICE for each object and role separately.

xiong-gang commented on src/backend/catalog/aclchk.c:
```diff
@@ -486,9 +491,37 @@ ExecuteGrantStmt(GrantStmt *stmt)
+	{
+		ereport(NOTICE,
+				(errcode(ERRCODE_WARNING_PRIVILEGE_NOT_REVOKED),
+				 errmsg("no privileges could be revoked")));
```
can we return from here?

hlinnaka replied:
Hmm, to skip the dispatching, if there was nothing to do? I suppose we could. Although perhaps better not to, just in case the QD node is out-of-sync with the segments, and there is something to revoke in the segments even though there was nothing in the master. This isn't performance-critical, after all. Also, for comparison, DROP IF EXISTS is dispatched to segments, even if the objects didn't exist in the master.

xiong-gang commented:
LGTM

hlinnaka commented:
Ok, then. I updated the expected output for the message change, and rebased. Consider this a concrete proposal, then, no longer work-in-progress. Assuming the PR pipeline is happy with this, I'll go ahead an push.

Yes, a hint or other clarification of the NOTICE would still be nice. That can be done as a separate commit, so I'm not holding off pushing this waiting for that. But please create a github issue or something to track it, I'm not going to follow-up on that myself.

gaos1 commented:
I created an issue to follow up the hint message: 
https://github.com/greenplum-db/gpdb/issues/4060
